components/service-operator/controllers/s3_cloudformation_test.go

```diff
@@ -0,0 +1,170 @@
+				},
+			}
+			secret core.Secret
+		)
+
+		By("creating a prequisite Principal resource with kubernetes api", func() {
+			Expect(client.Create(ctx, &principal)).To(Succeed())
+		})
+
+		By("creating an S3Bucket resource with kubernetes api", func() {
+			Expect(client.Create(ctx, &bucket)).To(Succeed())
+		})
+
+		By("displaying a READY resource status after initial creation", func() {
+			Eventually(func() object.State {
+				_ = client.Get(ctx, resourceNamespacedName, &bucket)
+				return bucket.GetState()
+			}, timeout).Should(Equal(object.ReadyState))
+		})
+
+		By("displaying an AWS CREATE_COMPLETE resource status after initial creation", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &bucket)
+				return bucket.Status.AWS.Status
+			}, timeout).Should(Equal(cloudformation.CreateComplete))
+		})
+
+		By("displaying an AWS stack id in resource status", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &bucket)
+				return bucket.Status.AWS.ID
+			}).ShouldNot(BeEmpty())
+		})
+
+		By("displaying a stack name prefixed with cluster name in resource status", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &bucket)
+				return bucket.Status.AWS.Name
+			}).Should(ContainSubstring("xxx-s3-test-test-bucket"))
+		})
+
+		By("ensuring a finalizer is present on resource to prevent deletion", func() {
+			Eventually(func() []string {
+				_ = client.Get(ctx, resourceNamespacedName, &bucket)
+				return bucket.Finalizers
+			}).Should(ContainElement(cloudformation.Finalizer))
+		})
+
+		By("ensuring no DeletionTimestamp exists", func() {
+			Eventually(func() bool {
+				_ = client.Get(ctx, resourceNamespacedName, &bucket)
+				return bucket.ObjectMeta.DeletionTimestamp == nil
+			}).Should(BeTrue())
+		})
+
+		By("creating a secret with bucket details", func() {
+			Eventually(func() map[string][]byte {
+				_ = client.Get(ctx, secretNamespacedName, &secret)
+				return secret.Data
+			}).Should(HaveKey("S3BucketName"))
+		})
+
+		By("creating a secret with the principal role name", func() {
+			Eventually(func() map[string][]byte {
+				_ = client.Get(ctx, secretNamespacedName, &secret)
+				return secret.Data
+			}).Should(HaveKey("IAMRoleName"))
+		})
+
+		By("deleteing S3Bucket resource with kubernetes api", func() {
+			err := client.Get(ctx, resourceNamespacedName, &bucket)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(client.Delete(ctx, &bucket)).To(Succeed())
+		})
+
+		By("ensuring the S3Bucket resources have been removed", func() {
+			Eventually(func() int {
+				var list storage.S3BucketList
+				err := client.List(ctx, &list)
+				Expect(err).ToNot(HaveOccurred())
+				return len(list.Items)
+			}, timeout).Should(Equal(0))
+		})
+
+		// By("ensuring secret has been removed", func() {
+		// 	var secretList core.SecretList
+		// 	Eventually(func() int {
+		// 		err := client.List(ctx, &secretList)
```

Comment:
Can we remove the commented out test case?

Reply:
I would like to leave it in, because it really should be fixed and not forgotten about